File: src/main/java/bad/robot/radiate/ui/FrameRate.java
package bad.robot.radiate.ui;

class FrameRate {

    public static final FrameRate videoFramesPerSecond = framesPerSecond(24);

    private final int framesPerSecond;

    static FrameRate
    framesPerSecond(int framesPerSecond) {
        return new FrameRate(framesPerSecond);
    }

    private FrameRate(int framesPerSecond) {
        if (framesPerSecond < 1)
            throw new IllegalArgumentException("frame rates less than 1 seems a little low");
        if (framesPerSecond > 60)
            throw new IllegalArgumentException("frame rates over 60 frames per seconds seems a little high");
        this.framesPerSecond = framesPerSecond;
    }

    int asFrequencyInMillis() {
        return 1000 / framesPerSecond;
    }
}

File: src/main/scala/bad/robot/radiate/ui/FrameRate.scala
package bad.robot.radiate.ui

object FrameRate {
  val videoFramesPerSecond: FrameRate = framesPerSecond(24)

  private[ui] def framesPerSecond(framesPerSecond: Int): FrameRate = {
    new FrameRate(framesPerSecond)
  }
}

class FrameRate(framesPerSecond: Int) {
  if (framesPerSecond < 1) throw new IllegalArgumentException("frame rates less than 1 seems a little low")
  if (framesPerSecond > 60) throw new IllegalArgumentException("frame rates over 60 frames per seconds seems a little high")

  private[ui] def asFrequencyInMillis: Int = 1000 / framesPerSecond
}